src/PrestaShopBundle/Controller/Admin/MultistoreController.php

```diff
@@ -115,20 +115,21 @@ public function header(bool $lockedToAllShopContext): Response
      */
     public function configurationDropdown(ShopConfigurationInterface $configuration, string $configurationKey): Response
     {
-        $groupList = $this->entityManager->getRepository(ShopGroup::class)->findBy(['active' => true]);
+        $shopGroups = $this->entityManager->getRepository(ShopGroup::class)->findBy(['active' => true]);
         $shopCustomizationChecker = $this->get('prestashop.multistore.customized_configuration_checker');
         $isGroupShopContext = $this->multistoreContext->isGroupShopContext();
         $shouldDisplayDropdown = false;
+        $groupList = [];
 
-        foreach ($groupList as $key => $group) {
-            if (count($group->getShops()) === 0) {
-                unset($groupList[$key]);
+        foreach ($shopGroups as $key => $group) {
+            if ($this->shouldIncludeGroupShop($group)) {
+                $groupList[] = $group;
             }
-            if ($this->multistoreContext->isAllShopContext() || $group->getId() === $this->multistoreContext->getContextShopGroup()->id) {
+            if ($this->multistoreContext->isAllShopContext() || $group->getId() === $this->multistoreContext->getContextShopGroup()->id && !$shouldDisplayDropdown) {
```

Comment:
Hello @matthieu-rolland . I understand that what you are doing in function `configurationDropdown()` is

1. Gather multishop data
2. Do some computing and filtering to build $groupList
3. Return the dropdown, with the right data to be displayed

The step 2 is being run when the shop context is all shops, group shop and single shop. The 3 usecases run the same code which is why you have some conditional statement "if I am in shop group", "if I am in single shop"...

I may be out of scope, but I was wondering if the code could be easier to navigate if we handle the 3 usecases very differently?

I imagine something like this:

```
    public function configurationDropdown(ShopConfigurationInterface $configuration, string $configurationKey): Response
    {
        switch ($contextType)
            case self::CONTEXT_ALL_SHOPS:
                return $this->build AllShopsDropdown();
            case self::CONTEXT_GROUP_SHOPS:
                return $this->build GroupShopsDropdown();
            case self::CONTEXT_SINGLE_SHOP:
                return $this->build SingleShopDropdown();
    }
```

Then all the conditional logic "should I include group shop? yes? no?" would not be needed. For the 3 functions you would know exactly what to do.

Reply:
Actually there is only two cases: all shop or group shop (we never display this dropdown in single shop context, so this method is not even called).

I just separated the data gathering into two methods, one for all shop, and one for group shop, there is a little bit of repetition, but it's easier to read. 

If you guys think it was better before, I can still remove this last commit 👍